plugins/modules/monitoring/newrelic_deployment.py

```diff
@@ -15,23 +15,18 @@
 author: "Matt Coddington (@mcodd)"
 short_description: Notify newrelic about app deployments
 description:
-   - Notify newrelic about app deployments (see https://docs.newrelic.com/docs/apm/new-relic-apm/maintenance/deployment-notifications#api)
+   - Notify newrelic about app deployments (see https://docs.newrelic.com/docs/apm/new-relic-apm/maintenance/record-monitor-deployments/)
 options:
   token:
     type: str
     description:
-      - API token, to place in the x-api-key header.
+      - API token, to place in the Api-Key header.
     required: true
-  app_name:
```

Comment:
Usually removing options isn't allowed, since it's a breaking change. In the situation we are in right now - this module doesn't work at all anymore - we might be able to ignore it, but we should only do it if it's impossible to use this option with the new API. The code in https://github.com/ansible-collections/community.general/pull/1501/files#diff-09407ffe9247260fa13957bfac792dedb8889dd9e970b399aacb206267eb6583R119 *seems* to indicate that this can still be made to work. If that's true, then this option should stay.

Reply:
Thanks for the review @felixfontein! Right, v2 is a breaking change itself, so ``appname``, ``app_name``, ``environment`` and ``validate_certs`` are basically no longer needed.

I took a look at that PR you linked, maybe it worked 2 years ago, but not sure the same would apply now as the API specifically is looking for the ``application_id``. At least I don't see in the official documentation that says otherwise.

Reply:
The code from that PR lists all deployments filtered by that application name, and it finds one it uses its application ID. Assuming the list with filtering option is still available, I guess it should still work (and it sounds reasonable to me that this is still available).

Reply:
Ah yea I see, it does sound reasonable, I'll check it out thanks.